src/cryptonote_core/blockchain.cpp

```diff
@@ -4142,6 +4160,7 @@ void Blockchain::block_longhash_worker(uint64_t height, const epee::span<const b
     if (m_cancel)
        break;
     crypto::hash id = get_block_hash(block);
+    setup_seedhash(block, height);
     crypto::hash pow = get_block_longhash(block, height++);
```

Comment:
If you're going to have to call setup_seedhash for every call, it should be moved into get_block_longhash since this does not require extra parameters.

Reply:
I didn't see a way to get to get_block_id_by_height() from inside get_block_longhash(). Maybe I missed something obvious.

Reply:
Oh, that accesses core ? That's unfortunate. It does feel like someday some call will be forgot though, there really should be a function that does both somewhere.

Reply:
Would be ideal to just move get_block_longhash() to core, instead of cryptonote_format_utils. What does it even have to do with "formatting" ?

Reply:
and you're right, looks like I missed an invocation in rpc/core_rpc_server.cpp. Not sure what that needs yet.

Reply:
Sounds good. Looking at current calls, nothing seems to be unable to use core for this.

Reply:
Ah this doesn't quite work, because miner::find_nonce_for_given_block is a static, doesn't have access to core. Not sure why that's that way either.

Reply:
It appears to be used only for creating the genesis block, and for some of the tests.